conans/client/conf/detect.py

```diff
@@ -64,10 +64,15 @@ def _clang_compiler(output, compiler_exe="clang"):
         installed_version = re.search("([0-9]+\.[0-9])", out).group()
         if installed_version:
             output.success("Found %s %s" % (compiler, installed_version))
+            major = installed_version.split(".")[0]
```

Comment:
I am afraid this would be a breaking change, what do you think @lasote?

Reply:
It's only for Clang and version >= 7.
What do you fear that this would break?

Reply:
If you are trusting your default profile and you have packages generated for "7.0" It won't find the "7" packages anymore.

Reply:
how does it work for GCC 5.0 vs GCC 5 for instance?

Reply:
at least I didn't spot any GCC-specifc quirks in ConanInfo implementation, could you point me?

Reply:
What about changing it only for Clang >= 8 then and allowing via default Conan setting.yml both values 7 and 7.0 for the moment?

This could be left this way or changed retrospectively with Conan 2.0

Since Clang 8 is not releases yet it wouldn't break existing behaviour.

Reply:
I agree with Croydon
how about changing it for Clang >= 8.0, and use docker images `conan/lasoteclang70` but `conan/lasoteclang8`?

Reply:
after the today's discussion, we decided:
- [x] keep Clang 7.0 to avoid compatibility breakage
- [x] since Clang 8, use major only
- [x] for docker images, use `conanio/clang70` and `conanio/clang8`, for the consistency
pull request has been updated accordingly, see notes below